querybook/server/lib/ai_assistant/base_ai_assistant.py

```diff
@@ -1,12 +1,18 @@
 from abc import ABC, abstractmethod
-
+import functools
 import queue
 
 from langchain.callbacks.streaming_stdout import StreamingStdOutCallbackHandler
+from pydantic.error_wrappers import ValidationError
+
+from lib.logger import get_logger
 
+LOG = get_logger(__file__)
 
-class ThreadedGenerator:
-    """Generator to facilitate streaming result from Langchain."""
+
+class EventStream:
```

Comment:
is this event stream generic? if so, can you put this into lib or datasource/?

Reply:
It'll be only needed by the ai assistant for now. I'd keep it here and move them when other use cases come up in the future.